### PR TITLE
KEYCLOAK-19503 escape name of the secret for client

### PR DESCRIPTION
**Pull request by jonathanvila**
Signed-off-by: Jonathan Vila <jvilalop@redhat.com>

## JIRA ID

https://issues.redhat.com/browse/KEYCLOAK-19503

## Additional Information

As the ticket says, when you use a clientId including a https url, it will fail as there are some chars not allowed for a K8s resource name

## Verification Steps
1. make cluster/prepare
2. Run operator from IDE , *from master branch*
2. deploy a keycloak.yaml
3. edit the keycloak CR and delete the annotation "service.alpha.openshift.io/serving-cert-secret-name"
4. delete the x509 secret
5. deploy a keycloakrealm.yaml
6. try to deploy a keycloakclient with clientid "https://whatever?value"  ---> will fail
7. stop the operator on the IDE, and now go to the PR branch, and run it
8. try to deploy a keycloakclient with clientid "https://whatever?value"  ---> will work

## Checklist:
- [X] Automated Tests
- [ ] [Documentation](https://github.com/keycloak/keycloak-documentation) changes if necessary
- [ ] (If required) Discussed on [Keycloak DEV](https://groups.google.com/forum/#!forum/keycloak-dev)

### CODE REVIEW

File: pkg/model/client_secret.go [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
+	// we guarantee first and last char is an alpha and not a dot or a dash
+	if !unicode.IsLetter(reg[0]) && !unicode.IsNumber(reg[0]) {
+		reg = append([]rune{'a'}, reg...)
+	}
+	if !unicode.IsLetter(reg[len(reg)-1]) && !unicode.IsNumber(reg[len(reg)-1]) {
+		reg = append(reg, 'a')
+	}
+
+	return string(reg)
+}
```
Comment (vmuzikar) [inline pkg/model/client_secret.go:26]: Shouldn't we move this to the util file and rename it to something like `SanitizeResourceNameWithAlphaNum`? Not sure about the name, though...

File: pkg/model/client_secret.go [hunk 2]
```diff
@@ -24,8 +43,9 @@ func ClientSecret(cr *v1alpha1.KeycloakClient) *v1.Secret {
 }
 
 func ClientSecretSelector(cr *v1alpha1.KeycloakClient) client.ObjectKey {
+	escapedClientIDName := SanitizeResourceName(cr.Spec.Client.ClientID)
```
Comment (vmuzikar) [inline pkg/model/client_secret.go:46]:
```suggestion
	escapedClientIDName := MakeK8sCompatibleName(cr.Spec.Client.ClientID)
```

**Review by vmuzikar — CHANGES_REQUESTED** [review]

File: pkg/model/client_secret.go (hunk 1 above)
Comment (jonathanvila) [reply pkg/model/client_secret.go:26]: OK

File: pkg/model/client_secret.go (hunk 2 above)
Comment (jonathanvila) [reply pkg/model/client_secret.go:46]: :+1:

**Review by vmuzikar — APPROVED** [review]
@jonathanvila LGTM, thank you!

**Pull request merged**